src/python/gem5/resources/client_api/atlasclient.py

```diff
@@ -125,6 +126,27 @@ def _atlas_http_json_req(
         """
         data = json.dumps(data_json).encode("utf-8")
 
+        # check to see if user requests a proxy connection
+        use_proxy = os.getenv("GEM5_USE_PROXY")
+        if use_proxy:
+            # If the "use_proxy" variable is specified we setup a socks5
+            # connection.
```

Comment:
Does it make sense to split this out into a function? Is this duplicated code from the other place you check proxies?